config/application.rb

```diff
@@ -72,20 +72,265 @@ class Application < Rails::Application
+          'add it to the config/settings.local.yml (secret_token: your_token).'\
+          ' To generate one, the following command can be used: ' \
+          '"openssl rand -hex -rand /var/log/messages 64 2>/dev/null".'
+      end
+    end
+
+    Settings.email = "noreply@#{config.fqdn}" unless Settings.email
+    config.email = Settings.email
 
+    # init Devise.mailer.*
+    s = Settings.action_mailer
+    ex = config.action_mailer
+    ex.default_url_options = { :host => config.fqdn, :port => config.port }
+    if ! s
+      ex.perform_deliveries = Rails.env != 'test'
+      ex.raise_delivery_errors = Rails.env != 'production'
+    else
+      ex.perform_deliveries = s.perform_deliveries \
+        ? s.perform_deliveries.to_s == 'true'
+        : Rails.env != 'test'
+      ex.raise_delivery_errors = s.raise_delivery_errors \
+        ? s.raise_delivery_errors.to_s == 'true'
+        : Rails.env != 'production'
     end
+    if ! (s && s.smtp_settings)
+      ex.smtp_settings = {
+        # the Devise.mailer defaults do NOT make sense
+        :address => 'mail',
+        :port => 25,
+        :domain => nil,
+        :enable_starttls_auto => true,
+        :password => nil,
+        :authentication => nil,
+      }
+    else
+      s = s.smtp_settings
+      tmp = s.address ? s.address.to_s : ''
+      ex.address = tmp.length > 0 ? tmp : 'mail'
+      i = s.port ? s.port.to_i : 0
+      ex.port = i > 0 ? i : 25
+      tmp = s.domain ? s.domain.to_s : ''
+      ex.domain = tmp.length > 0 ? tmp : nil
+      tmp = s.enable_starttls_auto ? s.enable_starttls_auto.to_s : ''
+      ex.enable_starttls_auto = tmp == 'true'
+      ex.password = s.password.to_s if s.password
+      ex.authentication = s.authentication.to_s if s.authentication
+    end
+    tmp = (s && s.delivery_method) ? s.delivery_method.to_s : ''
+    ex.delivery_method = (tmp.in? ['sendmail', 'smtp', 'file', 'test']) \
+      ? tmp.intern
+      : 'sendmail'.intern
+
+    tmp = Settings.name ? Settings.name : ''
+    config.name = tmp.length < 1 ? 'MyOntohub' : tmp
+	Settings.name = config.name
+
+    tmp = Settings.asset_host ? Settings.asset_host.to_s : ''
+    config.action_controller.asset_host = tmp if tmp.length > 0
+
+    config.display_head_commit = Settings.display_head_commit \
+      ? Settings.display_head_commit.to_s == 'true'
+      : Rails.env != 'production'
+    Settings.display_head_commit = config.display_head_commit
+
+	Settings.display_symbols_tab =
+      Settings.display_symbols_tab && Settings.display_symbols_tab.to_s=='true'
+
+    Settings.format_selection =
+      Settings.format_selection && Settings.format_selection.to_s == 'true'
+
+    if ! Settings.footer
+      # only if nil!
+      Settings.footer = [
+        OpenStruct.new('text'=>'Imprint', 'href'=>'http://about.ontohub.org/'),
+        OpenStruct.new('text'=>'Source Code',
+          'href'=>'https://github.com/ontohub/ontohub')
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Surrounding space missing for operator '=>'.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.